osu.Game.Rulesets.Taiko/Mods/TaikoModAlternate.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.Taiko.Objects;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public class TaikoModAlternate : ModAlternate<TaikoHitObject, TaikoAction>
+    {
+        private TaikoAction? lastRimAction;
+        private TaikoAction? lastCentreAction;
+
+        protected override void OnBreakEnd()
+        {
+            resetStates();
+        }
+
+        protected override void OnInterceptorLoadComplete()
+        {
+            Drawable interceptor = Interceptor as Drawable;
+
+            using (Interceptor.BeginAbsoluteSequence(0))
+            {
+                foreach (TaikoHitObject hitObject in HitObjects)
+                {
+                    if (!hitObject.IsStrong) continue;
+
+                    var window = hitObject.HitWindows.WindowFor(HitResult.Miss);
+                    interceptor.Delay(hitObject.StartTime - window).Schedule(() => resetStates());
+                    interceptor.Delay(hitObject.StartTime + window).Schedule(() => resetStates());
```

Comment:
Not sure this is completely correct; what if there are two objects with overlapping hit windows?

Reply:
It should be intended for drum rolls only. I'll add a type check before that.